server/rest/src/main/java/org/infinispan/rest/resources/ContainerResource.java

```diff
@@ -372,6 +382,50 @@ private CacheInfo getCacheInfo(RestRequest request, EmbeddedCacheManager cacheMa
       return cacheInfo;
    }
 
+   private CompletionStage<RestResponse> getCacheNamesPerRole(RestRequest request) throws RestResponseException {
+      NettyRestResponse.Builder responseBuilder = checkCacheManager(request);
+      if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
+
+      String roleName = request.getParameter("roleName");
+      NettyRestResponse.Builder builder = invocationHelper.newResponse(request);
+
+      if (roleName == null) {
+         return completedFuture(builder.status(HttpResponseStatus.BAD_REQUEST).build());
+      }
+
+      GlobalAuthorizationConfiguration authorization = invocationHelper.getRestCacheManager()
+            .getInstance().getCacheManagerConfiguration().security().authorization();
+      Role role = authorization.getRole(roleName);
+
+      if (role == null) {
+         return completedFuture(builder.status(NOT_FOUND).build());
+      }
+
+      // secured caches
+      Set<String> cachesNames = new HashSet<>(invocationHelper.getRestCacheManager().getAccessibleCacheNames());
+      cachesNames.removeAll(internalCacheRegistry.getInternalCacheNames());
+
+      List<Set<String>> securedAndNotSecured = List.of(new HashSet<>(), new HashSet<>());
```

Comment:
Why are you putting these in a list ?

Reply:
no reason for this code, I  was working on a method return first